File: in_Java/0203 Remove Linked List Elements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        if(head == null) return head;
        while(head != null && head.val == val) head = head.next;
        ListNode d = head;
        while(head != null && head.next != null){
            if(head.next.val == val) head.next = head.next.next;
            else head = head.next;
        }
        return d;
    }
}

File: in_Python/0203 Remove Linked List Elements.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        pre = ListNode()
        pre.next = head
        nh = pre
        while head != None:
            if head.val == val:
                head = head.next
                pre.next = head
            else:
                pre = head
                head = head.next
        return nh.next
        